### DISCUSSION TITLE
Implement ReactPrime (.fig) design system when a design system already exists in the project.

### DISCUSSION THREAD

**Discussion by MauricioUX** [primefaces/community#2003] [Figma UI Kit] [answered]
Hi!
I work in a software development company and I'm working on a large project as a UX UI designer. The project is a year and a half old, with the first move to production earlier this year.

I joined the team along with the move to production as the project continues to grow with new challenges.

The design system was built by hand with all its components, styling, etc. We fell short with that design system, along with some standardisation issues between the design team and the frontend development team.

One of the senior developers proposed to implement ReactPrime into the project, which was well received. Then, from my side, as a designer, the challenge arose of how to implement the ReactPrime library, with the current design system, in such a way, to justify the purchase of the license you sell (figma file).

So, my questions are:
1. Considering that today we have in Figma a custom, handmade design system, what is the best practice to implement ReactPrime (Figma) in our project? I make the scope, that for the moment will coexist in the final product, both design systems, as there are things (components) that have been created exclusively for this project. I think we will be replacing little by little the more general components that currently exist, for example, calendars, input, among others. Do we take our current design system to a copy of the ReactPrime design system, and there adjust the styles and configurations of this project or take from ReactPrime, only the components that we will eventually occupy and copy them to our current design system and there we adjust the styles?

2. What happens with the ReactPrime license (.fig) in that case? In order for the purchase to be justified in the company, we want to implement this new design system in this project as well as in future projects. So, part of our deliverables to the customer is the product design file and we also deliver the design system. So, my question is: With the license we would buy, how can we sell our service considering the above.

Please, I need the answer as detailed as possible, since we must get out of doubts and justify the purchase to my employer.

I look forward to your comments. Thank you!
Mauricio

**Comment by onursenture** [comment, marked as answer]
Hello,

As the name suggests, PrimeOne UI Kit is a UI Kit that includes all possible variants of the components available in the PrimeReact library. It is a Figma library that ensures elements like colors and padding comply with our frontend libraries within a predefined system.

Your situation is quite common among PrimeOne UI Kit users. Many companies already have their own design systems in place. Despite this, PrimeReact is often chosen because it dramatically simplifies the frontend library side of things, making it challenging to adapt the existing design system to fit it.

Personally, I haven't conducted such an adaptation for another company, so my advice doesn't come from direct experience. However, if I were to face such a situation, here's what I would do: I would use the PrimeOne UI Kit as a starting point and then apply the components from your existing design system to it, ordering them from most to least used. Throughout this process, I would document every step. While we typically avoid frequent updates, documenting the steps will be helpful if you need to implement the same changes after a Prime UI Kit update from PrimeTek, which may include breaking changes.

This way, by making custom changes in PrimeReact from the starting point we provided, developers can more easily implement the transition.

Regarding handing over the design file to a customer: if you are giving PrimeOne UI Kit to the customer 'as is,' you'll need to purchase a license for each customer you provide it to. However, if the file has been entirely modified, transferring this file to the customer does not require a license. For more detailed information about licensing, you can email your legal concerns to contact 🅰️ primetek.com.tr

**Comment by MauricioUX** [comment]
Thank you!
